google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -2154,6 +2157,80 @@ public ApiFuture<Empty> asyncClose() {
     }
   }
 
+  private final class MultiplexedSessionWaiterFuture
+      extends ForwardingListenableFuture<MultiplexedSession> {
+
+    private final Duration MAX_MULTIPLEXED_SESSION_WAIT_TIMEOUT = Duration.ofSeconds(60);
+    private final SettableFuture<MultiplexedSession> waiter = SettableFuture.create();
+
+    @Override
+    protected ListenableFuture<? extends MultiplexedSession> delegate() {
+      return waiter;
+    }
+
+    private void put(MultiplexedSession session) {
+      waiter.set(session);
+    }
+
+    private void put(SpannerException e) {
+      waiter.setException(e);
+    }
+
+    @Override
+    public MultiplexedSession get() {
+      ISpan span = tracer.spanBuilder(WAIT_FOR_MULTIPLEXED_SESSION);
+      try (IScope waitScope = tracer.withSpan(span)) {
+        final Duration timeout =
+            MoreObjects.firstNonNull(
+                options.getAcquireSessionTimeout(), MAX_MULTIPLEXED_SESSION_WAIT_TIMEOUT);
+        return pollUninterruptedlyWithTimeout(timeout);
+      } catch (Exception e) {
+        if (e instanceof SpannerException
+            && ErrorCode.DEADLINE_EXCEEDED.equals(((SpannerException) e).getErrorCode())) {
+          numMultiplexedSessionWaiterTimeouts.incrementAndGet();
+          tracer.getCurrentSpan().setStatus(ErrorCode.DEADLINE_EXCEEDED);
+        }
+        span.setStatus(e);
+        throw e;
+      } finally {
+        span.end();
+      }
+    }
+
+    /**
+     * Method which allows to obtain a multiplexed session after blocking for a configurable
+     * duration {@link SessionPoolOptions#getAcquireSessionTimeout()}. Note that this duration
+     * becomes obsolete in case we have set {@link
+     * SessionPoolOptions#getWaitForMultiplexedSession()}. Because {@link
+     * SessionPoolOptions#getWaitForMultiplexedSession()} will ensure that the multiplexed session
+     * is initialized and available during application start up.
+     */
+    private MultiplexedSession pollUninterruptedlyWithTimeout(Duration acquireSessionTimeout) {
```

Comment:
This poll method made sense for regular sessions, because those could be blocked during normal application operation, waiting for a session to be returned by another thread. Now, waiting will only potentially happen at startup while `CreateSession` is in flight, and will never be longer than the timeout value for the `CreateSession` RPC (which by default is lower than the fixed timeout value that is used here).